src/components/InputTime/AsDate.tsx

```diff
@@ -0,0 +1,162 @@
+import React, {
+  Fragment,
+  MutableRefObject,
+  useEffect,
+  useImperativeHandle,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+
+import {
+  getEndOfDay,
+  getShortTimeString,
+  getStartOfDay,
+  handleDispatchNativeInputChange,
+} from './utils';
+
+import InputTimeAsString, { AsStringProps } from './AsString';
+
+export const AsDate: React.FC<AsStringProps> = ({
+  disabled = false,
+  forwardedRef,
+  max: maxDateString,
+  min: minDateString,
+  onChange,
+  value: valueDateString,
+  ...passedProps
+}) => {
+  const [invalidMax, setInvalidMax] = useState(false);
+  const [invalidMin, setInvalidMin] = useState(false);
+
+  useEffect(() => {
+    if (maxDateString && Number.isNaN(new Date(maxDateString).valueOf())) {
+      console.warn('`max` should be an ISO date string');
```

Comment:
One quick thought: it might be useful to add a reference to `InputTime`. May help a bit in tracking down where the error is originating